File: lib/domain/cultivation_plan/mappers/public_plan_save_session_data_mapper.rb
# frozen_string_literal: true

module Domain
  module CultivationPlan
    module Mappers
      class PublicPlanSaveSessionDataMapper
        # @param header [Dtos::PublicPlanSaveHeaderSnapshot]
        # @param field_rows [Array<Dtos::PublicPlanSaveFieldDatum>]
        # @return [Dtos::PublicPlanSaveSessionData]
        def self.from_snapshots(header:, field_rows:)
          Dtos::PublicPlanSaveSessionData.new(
            plan_id: header.plan_id,
            farm_id: header.farm_id,
            field_data: field_rows
          )
        end
      end
    end
  end
end

File: crates/agrr-domain/src/cultivation_plan/mappers/public_plan_save_session_data_mapper.rs
//! Ruby: `Domain::CultivationPlan::Mappers::PublicPlanSaveSessionDataMapper`

use crate::cultivation_plan::dtos::{
    PublicPlanSaveFieldDatum, PublicPlanSaveHeaderSnapshot, PublicPlanSaveSessionData,
};

pub fn from_snapshots(
    header: &PublicPlanSaveHeaderSnapshot,
    field_rows: &[PublicPlanSaveFieldDatum],
) -> PublicPlanSaveSessionData {
    PublicPlanSaveSessionData::new(
        header.plan_id,
        header.farm_id,
        field_rows.to_vec(),
        None,
    )
}

#[cfg(test)]
mod mappers_public_plan_save_session_data_mapper_test_inline {
    use super::*;
    include!(concat!(env!("CARGO_MANIFEST_DIR"), "/test/cultivation_plan/mappers_public_plan_save_session_data_mapper_test.rs"));
}
